guides/attachvolume.mdx

```diff
@@ -0,0 +1,197 @@
+
+---
+
+### Step 2: Create a Volume
+
+Now that you know the volume type and region, create the volume using the [`/volumes/create`](https://docs.shadeform.ai/api-reference/volumes/volumes-create) API.
+
+<Note> Your volume and instance will need to have the same cloud provider, and be in the same region to work </Note>
+
+Replace `<api-key>` with your API key, and customize the `cloud`, `region`, and `size_in_gb` fields to match your requirements.
+
+<CodeGroup>
+  ```bash CLI
+  # Create a new volume
+  curl --location 'https://api.shadeform.ai/v1/volumes/create' \
+  --header 'x-api-key: <api-key>' \
+  --header 'Content-Type: application/json' \
+  --data '{
+      "cloud": "digitalocean",
+      "region": "tor1",
+      "size_in_gb": 100,
+      "name": "mystoragevolume"
+  }'
+  ```
+</CodeGroup>
+
+**Example Response**
+
+The response will include the `id` of the newly created volume, which will be used to attach it to an instance.
+
+```json Response
+{
+  "id": "df5c952c-0567-4140-94d6-59cee96f8caa"
+}
+```
+
+Save the `id` field as you will need it in the next step.
+
+---
+
+### Step 3: Create an Instance with the Volume Attached
+
+To attach the volume, pass its `id` in the `volume_ids` field while creating the instance using the [`/instances/create`](https://docs.shadeform.ai/api-reference/instances/instances-create) endpoint. 
+
+<Note> "volume_ids" must be an array of size one. This is because we only allow attaching one volume to one instance right now. Attaching multiple volumes to a single instance is a feature we may add in the future. </Note>
+
+<CodeGroup>
+  ```bash CLI
+  # Create an instance with the volume attached
+  curl --location 'https://api.shadeform.ai/v1/instances/create' \
+  --header 'x-api-key: <api-key>' \
+  --header 'Content-Type: application/json' \
+  --data '{
+    "cloud": "digitalocean",
+    "region": "tor1",
+    "shade_instance_type": "H100_sxm5",
+    "shade_cloud": true,
+    "name": "the-super-cool-digitalocean-server-volume",
+    "volume_ids": ["b1f2c5d3-29ab-4e7b-9f7e-2c35a1e34d4e"] # be sure to include this line
+  }'
+  ```
+</CodeGroup>
+
+**Example Response**
+
+The response will include the `id` for the instance and a `cloud_assigned_id`. Save this for further operations.
+
+```json Response
+{
+  "id": "cc9f6b74-9825-4854-9e9c-dd50c7e97c3a",
+  "cloud_assigned_id": "720f2a6a-e4ee-488a-ade1-e7892f5d730a"
+}
+```
+
+---
+
+### Step 4: Verify the Volume on the Instance
+
+Once the instance is active, SSH into it to verify that the volume has been mounted. Learn how to SSH into a Shadeform instance using [this guide](https://docs.shadeform.ai/getting-started/quickstart#sshing-into-the-instance).
+
+After connecting, run the following command:
+
+```bash
+# List all block devices attached to the instance
+lsblk
+```
+
+This command will display all block devices attached to the instance. You should see your volume listed as a new device, such as `/dev/sdb` or similar.
+
```

Comment:
Can we add a new step here that is "Step 5: Mount the volume to a file system" and then show the commands:

sudo mkdir /dev/sdb
sudo mkfs.ext4 /dev/sdb
sudo mount -t ext4 /dev/sdb /mnt/sdb/
sudo chown -R shadeform:shadeform /mnt/sdb/